odoo/orm/models.py

```diff
@@ -3866,10 +3866,33 @@ def unlink(self) -> typing.Literal[True]:
         for sub_ids in split_every(cr.IN_MAX, self.ids):
             records = self.browse(sub_ids)
 
-            cr.execute(SQL(
-                "DELETE FROM %s WHERE id IN %s",
-                SQL.identifier(self._table), sub_ids,
-            ))
+            exc = False
+            # this try catch will only handle ForeignKeyViolation on delete
+            # There are no changes for any other exception raised during the delete or in a coming operation
+            try:
+                cr.execute(SQL(
+                    "DELETE FROM %s WHERE id IN %s",
+                    SQL.identifier(self._table), sub_ids,
+                ))
+            except psycopg2.errors.ForeignKeyViolation as exc:
```

Comment:
before my work, it was handeled here

https://github.com/odoo/odoo/blob/68bd44364de3128aac7252ae5e65a4ef97987a7a/odoo/orm/models.py#L2971-L2979

but at this level:
- i don't have the attributes like table name and sub_ids and i think it's not a good idea to extract them from the query text
- it's a place to handle the exception not to return the result

Reply:
@RaoufGhrissi I don't think we want/need it. A basic, generic, archive dialog in case of error when deleting is fine. And we already know in js if we're allowed to archive or not (see `activeActions`)

Reply:
Hello @aab-odoo  what if the error is not related to ForeignKeyViolation

in js i can't check the exception type as it's handeled in python (correct me if i'm wrong)

![image](https://github.com/user-attachments/assets/7cd579a9-88f8-4cf4-a0cb-50f091e713ac)

and i don't think it's a good idea  to extract it from the text.

Reply:
I think @ryv-odoo confirmed that catching UserError client side was good enough.

Reply:
okay, i will do it this way